include/sound/soc-acpi.h

```diff
@@ -61,6 +61,7 @@ static inline struct snd_soc_acpi_mach *snd_soc_acpi_codec_list(void *arg)
  * @platform: string used for HDAudio codec support
  * @codec_mask: used for HDAudio support
  * @common_hdmi_codec_drv: use commom HDAudio HDMI codec driver
+ * @sof_parent_device: use SOF configuration
```

Comment:
well the two cases are SOF or non-SOF. If there was a third option we'd use an enum but here it's not really necessary.

I also don't really like this too much, but I couldn't find a simple way to find what the parent device is. We can brainstorm a bit more on this, I agree this could be nicer.

Reply:
it's not bad @kv2019i. I only need this for ACPI cases, so it'd be useless indeed for the majority of PCI-based devices. We could have a static inline helper in this file so that there's only one place where we encode the driver name.